kv/dist_sender_test.go

```diff
@@ -752,3 +752,99 @@ func TestGetNodeDescriptor(t *testing.T) {
 		return util.Errorf("wanted NodeID 5, got %v", desc)
 	})
 }
+
+// TestSpanRangeScanWhenMerging simulate a range merge operation happend between get range desc
+// and get next range desc in getDescriptors function. the second sub scan use next range's start key
+// as start key of scan, this cause the the whole scan get duplicate key-values.
+func TestSpanRangeScanWhenMerging(t *testing.T) {
```

Comment:
comment suggestion:
> Test... simulates the situation in which the DistSender executes a multi-range scan which encounters the stale descriptor of a range which has since incorporated its right neighbor by means of a merge. It is verified that
the DistSender scans the correct keyrange exactly once.